tripy/tripy/frontend/tensor.py

```diff
@@ -106,52 +104,21 @@ def __init__(
 
         name = name if name is not None else Tensor._get_unique_name()
 
-        # set device now so we don't get errors due to not having a device attribute at all
-        # in the case where data is None
-        self.device = None
-
-        self.trace_tensor = TraceTensor(name, stack_info, None, None, None, None, shape=shape)
-
-        if isinstance(data, (List, tuple, bool, int, float)) and dtype is not None:
-            from tripy.frontend.trace.ops.cast import cast
-
-            element_type = get_element_type(data)
-            if element_type != dtype:
-                if element_type is not None:
-                    self.trace_tensor = cast(Tensor(data, shape, element_type, device), dtype=dtype).trace_tensor
-                    return
-                else:
-                    # We need explicit casting only if dtype can not be implicitly represented using `array.array`
-                    if dtype not in get_supported_array_type():
-                        from tripy.common.datatype import floating, integer
-
-                        element_type = dtype
-                        if issubclass(dtype, integer):
-                            element_type = tripy.common.datatype.int32
-                        elif issubclass(dtype, floating):
-                            element_type = tripy.common.datatype.float32
-                        self.trace_tensor = cast(Tensor(data, shape, element_type, device), dtype=dtype).trace_tensor
-                        return
+        self.trace_tensor = TraceTensor(name, stack_info, None, None, None, None)
 
         # Note: It is important that we are able to call the Tensor constructor with no arguments
         # since this is used internally.
-        if data is not None:
-            if not isinstance(data, Array):
-                data = Array(data, shape, dtype, device)
-            else:
-                # Internal usage only
-                # Disallow duplicate dtype/device when using Array to initialize a Tensor
-                if shape is not None:
-                    assert len(data.shape) == len(
-                        shape
-                    ), f"Rank provided to the initializer of Tensor (rank = {len(shape)}) does not match Array rank (rank = {len(data.shape)})."
-                assert not any([dtype, device]), "Duplicate arguments are not allowed. Use `Tensor(data)` instead."
-
-            # Data is present now. Assign the underlying device type.
-            self.device = data.device
-
-            Storage.build_internal([], [self.trace_tensor], data)
-            self.trace_tensor.shape = data.shape
+        if data is None:
+            return
+
+        if not isinstance(data, (Sequence, bool, int, float, runtime.MemRefValue)):
```

Comment:
Should we instead check whether `data` implements DLPack? What does `create_memref_view` require?